### PR TITLE
Add bruteForce solution (draft)

### PR DESCRIPTION
**Pull request by annnufan**
I'll make it better at the evening 
But main idea clear right now

### CODE REVIEW

File: group/assignment6/RearrangeCars.cpp [hunk 1]
```diff
@@ -139,3 +139,34 @@ void RearrangeCars::generateSequenceOfMovesWithConstraints(const vector<int> &en
...
 	}
 }
+
+void RearrangeCars::bruteForceAllSequence(vector<int>& currentStates, const vector<int>& endStates, vector<int>& carPosition, int emptySlotId, vector<Move>& sequenceOfMoves, vector<vector<Move>>& sequencesOfMoves, set<vector<int>>& usedPositions) {
+	int numOfCarsOnDisiredPosition = numberOfCarsOnDesiredPosition(carPosition, endStates);
+	if (numOfCarsOnDisiredPosition == carPosition.size()) {
+		sequencesOfMoves.push_back(sequenceOfMoves);
+		return;
+	}
+	for (int currCarSlotId = 0; currCarSlotId < carPosition.size(); currCarSlotId++) {
```
Comment (michbl) [inline group/assignment6/RearrangeCars.cpp]: I think there is a little problem here - there are N slots and the size of carPosition is N - 1. That means we will never get to the last slot. I guess we can change that to _currCarSlotID < currentStates.size();_

Comment (michbl) [inline group/assignment6/RearrangeCars.cpp]: I think there is a little problem here - there are N slots and the size of carPosition is N - 1. That means we will never get to the last slot. I guess we can change that to _currCarSlotID < currentStates.size();_

File: group/assignment6/RearrangeCars.cpp [hunk 2]
```diff
@@ -139,3 +139,34 @@ void RearrangeCars::generateSequenceOfMovesWithConstraints(const vector<int> &en
...
+		if (currCarSlotId != emptySlotId) {
```
Comment (michbl) [inline group/assignment6/RearrangeCars.cpp]: I think we should add another check here - _if (currCarSlotId != emptySlotId && validMove(carId, emptySlotId, constraints))_, and then this solution also supports constraints

File: group/assignment6/RearrangeCars.cpp [hunk 3]
```diff
@@ -139,3 +139,34 @@ void RearrangeCars::generateSequenceOfMovesWithConstraints(const vector<int> &en
...
+			moveCar(currentStates, carPosition, currCarSlotId, emptySlotId);
+			if (usedPositions.find(carPosition) == usedPositions.end()) {
+				usedPositions.insert(carPosition);
+				sequenceOfMoves.push_back(Move(currentStates[emptySlotId], make_pair(currCarSlotId, emptySlotId)));
+				bruteForceAllSequence(currentStates, endStates, carPosition, currCarSlotId, sequenceOfMoves, sequencesOfMoves, usedPositions);
+				sequenceOfMoves.pop_back();
```
Comment (michbl) [inline group/assignment6/RearrangeCars.cpp]: This section was confusing to me - I understand that we swapped the car in _currCarSlotId_ with the "car" in _emptySlotId_, but it's really hard to understand that the values in _currentStates_ changed when you first read it. Maybe we can also swap the values in _currCarSlotId_ and _emptySlotId_ after we push the move to the sequence of moves, or at least add a comment that explains what's happening here. What do you think?

File: group/assignment6/RearrangeCars.cpp (hunk 1 above)
Comment (annnufan) [reply group/assignment6/RearrangeCars.cpp]: I'll check it right now. But as I think, we need to try all N positions

File: group/assignment6/RearrangeCars.cpp (hunk 2 above)
Comment (annnufan) [reply group/assignment6/RearrangeCars.cpp]: Yeah, it's easy and add functionality to solution

**Comment by annnufan** [comment]
About name.. I'm not sure/ Now I just try to use consistent name for clearly reading

**Pull request merged**